src/pages/home/HeaderView.js

```diff
@@ -45,28 +46,33 @@ const propTypes = {
         guideCalendarLink: PropTypes.string,
     }),
 
+    parentReportActions: PropTypes.objectOf(PropTypes.shape(reportActionPropTypes)),
+
     ...windowDimensionsPropTypes,
     ...withLocalizePropTypes,
 };
 
 const defaultProps = {
     personalDetails: {},
+    parentReportActions: {},
     report: null,
     account: {
         guideCalendarLink: null,
     },
 };
 
 const HeaderView = (props) => {
+    // TO DO: Replace with subscribing to specific action rather than all.
+    const parentReportAction = ReportUtils.getParentReportAction_DEV(props.parentReportActions, `${props.report.parentReportActionID}`);
     const participants = lodashGet(props.report, 'participants', []);
     const participantPersonalDetails = OptionsListUtils.getPersonalDetailsForLogins(participants, props.personalDetails);
     const isMultipleParticipant = participants.length > 1;
     const displayNamesWithTooltips = ReportUtils.getDisplayNamesWithTooltips(participantPersonalDetails, isMultipleParticipant);
+    const isThread = ReportUtils.isThread(props.report);
     const isChatRoom = ReportUtils.isChatRoom(props.report);
     const isPolicyExpenseChat = ReportUtils.isPolicyExpenseChat(props.report);
     const isTaskReport = ReportUtils.isTaskReport(props.report);
-    const title = ReportUtils.getReportName(props.report);
-
+    const title = isThread ? lodashGet(parentReportAction, ['message', 0, 'text'], '') : ReportUtils.getReportName(props.report);
```

Comment:
Btw, if you want you can make a ReportUtils.getTitle which will call ReportActionUtils.getParentReportAction and check if that or reportName should be returned. It could DRY the code and keep title logic in one spot.

Reply:
Yeah I think that's a good call